bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/discovery/GreeDiscoveryService.java

```diff
@@ -0,0 +1,139 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.gree.internal.discovery;
+
+import static org.openhab.binding.gree.internal.GreeBindingConstants.*;
+
+import java.io.IOException;
+import java.net.DatagramSocket;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.i18n.LocaleProvider;
+import org.eclipse.smarthome.core.i18n.TranslationProvider;
+import org.eclipse.smarthome.core.net.NetworkAddressService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.gree.internal.GreeException;
+import org.openhab.binding.gree.internal.GreeTranslationProvider;
+import org.openhab.binding.gree.internal.handler.GreeAirDevice;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link GreeDiscoveryService} implements the device discovery service. UDP broadtcast ius used to find the devices on
+ * the local subnet.
+ *
+ * @author Markus Michels - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.gree")
+public class GreeDiscoveryService extends AbstractDiscoveryService {
+    private static final int TIMEOUT_SEC = 10;
+    private final Logger logger = LoggerFactory.getLogger(GreeDiscoveryService.class);
+    private final GreeTranslationProvider messages;
+    private final String broadcastAddress;
+
+    private GreeDeviceFinder deviceFinder = new GreeDeviceFinder();
+
+    @Activate
+    public GreeDiscoveryService(@Reference NetworkAddressService networkAddressService,
+            @Reference LocaleProvider localeProvider, @Reference TranslationProvider i18nProvider) {
+        super(SUPPORTED_THING_TYPES_UIDS, TIMEOUT_SEC);
+        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+        messages = new GreeTranslationProvider(bundle.getBundleContext().getBundle(), i18nProvider, localeProvider);
+        String ip = networkAddressService.getConfiguredBroadcastAddress();
+        broadcastAddress = ip != null ? ip : "";
+    }
```

Comment:
This will make sure that background scanning properly starts since the `@Activate` annotation is no longer on the `activate` method.

```suggestion
    public GreeDiscoveryService(@Reference NetworkAddressService networkAddressService,
            @Reference LocaleProvider localeProvider, @Reference TranslationProvider i18nProvider,
            @Nullable Map<String, @Nullable Object> configProperties) {
        super(SUPPORTED_THING_TYPES_UIDS, TIMEOUT_SEC);
        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
        messages = new GreeTranslationProvider(bundle.getBundleContext().getBundle(), i18nProvider, localeProvider);
        String ip = networkAddressService.getConfiguredBroadcastAddress();
        broadcastAddress = ip != null ? ip : "";
        activate(configProperties);
    }
```

Reply:
I changed to
```
   @Activate
    public GreeDiscoveryService(@Reference NetworkAddressService networkAddressService,
            @Reference GreeTranslationProvider translationProvider, @Nullable Map<String, @Nullable Object> configProperties) {
        super(SUPPORTED_THING_TYPES_UIDS, TIMEOUT_SEC);
        messages = translationProvider;
        String ip = networkAddressService.getConfiguredBroadcastAddress();
        broadcastAddress = ip != null ? ip : "";
        activate(configProperties);
    }

    @Override
    protected void activate(@Nullable Map<String, @Nullable Object> configProperties) {
        super.activate(configProperties);
    }
```

Is this ok? still having the activate() method?

Reply:
That is fine

Reply:
Why do you want to keep the overriden `activate()`, only invoking the base class method?

Reply:
that was my question, so I could safely remove it?

Reply:
yes, you can safely remove it

Reply:
done